arangod/RocksDBEngine/RocksDBVPackIndex.cpp

```diff
@@ -807,24 +807,28 @@ namespace {
 
 Result RocksDBVPackIndex::update(transaction::Methods& trx, RocksDBMethods* mthds,
                                  LocalDocumentId const& oldDocumentId,
-                                 velocypack::Slice const& oldDoc,
+                                 velocypack::Slice const oldDoc,
                                  LocalDocumentId const& newDocumentId,
-                                 velocypack::Slice const& newDoc,
-                                 Index::OperationMode mode) {
+                                 velocypack::Slice const newDoc,
+                                 OperationOptions& options) {
   if (!_unique) {
     // only unique index supports in-place updates
     // lets also not handle the complex case of expanded arrays
     return RocksDBIndex::update(trx, mthds, oldDocumentId, oldDoc,
-                                newDocumentId, newDoc, mode);
+                                newDocumentId, newDoc, options);
   }
 
   if (!std::all_of(_fields.cbegin(), _fields.cend(), [&](auto const& path) {
     return ::attributesEqual(oldDoc, newDoc, path.begin(), path.end());
   })) {
     // we can only use in-place updates if no indexed attributes changed
     return RocksDBIndex::update(trx, mthds, oldDocumentId, oldDoc,
-                                newDocumentId, newDoc, mode);
+                                newDocumentId, newDoc, options);
   }
+// simon: isn't this the default with the new protocol
```

Comment:
I don't think so. Shouldn't we still produce a new revision id and thus LocalDocumentId for each new revision of the document?